Signal/src/AppDelegate.m

```diff
@@ -197,9 +197,10 @@ - (UIViewController *)loadingRootViewController
         [[UIStoryboard storyboardWithName:@"Launch Screen" bundle:nil] instantiateInitialViewController];
 
     BOOL shouldShowUpgradeLabel = NO;
-    NSString *previousVersion = AppVersion.instance.lastAppVersion;
+    NSString *previousVersion = AppVersion.instance.lastCompletedLaunchAppVersion;
     // We added a number of database views in v2.13.0.
-    if ([VersionMigrations isVersion:previousVersion atLeast:@"2.0.0" andLessThan:@"2.13.0"]) {
+    if ([TSAccountManager isRegistered] &&
```

Comment:
In YapDatabase, it's writes that are blocking, not reads. See `CONCURRENCY` section here: https://github.com/yapstudios/YapDatabase/wiki/What-is-YapDatabase